### PR TITLE
Provide ability to pass execution user from resources

### PR DESCRIPTION
**Pull request by rathourarv**
# Description

resources -

    1. database
    2. user
    3. extensions

background -

It may possible that the machine which is going to run the recipe does
not have postgres user. Chef will automatically run the command eith root user
if we don't not specify the user.
We will lose the ability of peer authentication if we don't specify the user
so wecan provide the user name in params if you need to run sql commands using
specific user.

## Issues Resolved

https://github.com/sous-chefs/postgresql/issues/642

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 This file is used to list changes made in the last 3 major versions of the postgresql cookbook.
 
+## v7.1.9 (2020-03-04)
```
Comment (damacus) [inline CHANGELOG.md:5]:
```suggestion
## Unreleased
```

Comment (damacus) [reply CHANGELOG.md:5]: Please don't version bump in the changelog, we'll deal with this in a later PR. Thank you! :)

File: README.md [hunk 2]
```diff
@@ -161,6 +161,7 @@ Name          | Types  | Description
 `extension`   | String | Name of the extension to install the database                                    | Name of resource | yes
 `version`     | String | Version of the extension to install                                              |                  | no
 `old_version` | String | Older module name for new extension replacement. Appends FROM to extension query |                  | no
+`system_user` | String | Define the user on the system to execute sql command                             | Not set          | no
```
Comment (damacus) [inline README.md:164]:
```suggestion
`system_user` | String | Define the user on the system to execute sql command                             |           | no
```

Comment (rathourarv) [reply README.md]: Currently, the default user for Postgres extension is `Postgres`. I just add the ability to provide a user. If it's not provided then default `Postgres` as before.

Comment (damacus) [reply README.md]: We leave this field blank to denote this

Comment (rathourarv) [reply README.md]: I am afraid that I could not understand what we are trying to denote here by not providing the info.

Comment (rathourarv) [reply README.md]: I have made the changes but still I would love to hear reason behind it.

**Pull request closed without merging**